### PR TITLE
feat: Bedrock at AWS re:Invent 2023

### PR DESCRIPTION
**Pull request by Eriwyr**
Our feedback from aws reivent las vegas

### CODE REVIEW

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+title: Bedrock at 2023 AWS re:Invent Las Vegas
+description: A look back at the 2023 edition of the Las Vegas AWS re:Invent
+author: [a_ferez, v_chabrier]
+tags: [kubernetes, cloud, devops, opensource, community, conference, aws, re:invent, las vegas, 2023]
+color: rgb(251,87,66)
+thumbnail: "/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-cover.jpeg"
+---
+
+
+AWS re:Invent 2023 was a showcase for GenAI. It was the announcement of Amazon Q, Amazon's new AI assistant, that attracted the most interest. Designed to meet the specific needs of businesses.
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:12]: AWS re:Invent 2023 was a showcase for GenAI. It was the announcement of Amazon Q, Amazon's new AI assistant, that attracted the most interest, designed to meet the specific needs of businesses.

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+Alongside these announcements, we had the opportunity to attend talks by some of the major players in streaming, such as Prime Video, Peacock, Netflix and Spotify. Their presentations offered valuable insights into their challenges, successes and lessons learned, enriching our own understanding of the sector.
+
+
+
+
+## How Amazon scales resilience to new heights 
+
+![Practice like you play](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-resilience.jpg)
+
+Olga Hall, Director of Live Events Availability & Resilience at Amazon Prime Video, and Lauren Domb,Chief Technologist, WWP Federal Financial Services, WW Chaos Engineering Lead at AWS, [discussed the importance of resilience in the streaming industry](https://www.youtube.com/watch?v=r3J0fEgNCLQ).
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:23]: Olga Hall, Director of Live Events Availability & Resilience at Amazon Prime Video, and Lauren Domb, Chief Technologist, WWP Federal Financial Services, WW Chaos Engineering Lead at AWS, [discussed the importance of resilience in the streaming industry](https://www.youtube.com/watch?v=r3J0fEgNCLQ).

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+He hadn't anticipated customer interest in a particular product. They had a massive demand for a product display and the service ended up being unavailable. At this point, traffic increased again sharply as customers began reloading the page. Above all, it highlights the fact that, as their services are slow to respond, the answers are no longer relevant, leaving us with a system that is 100% loaded and no longer doing anything useful.
+
+These examples show that congestion collapse is a real problem that requires constant attention and planning. 
+
+At bedrock we've also encountered congestion collapse. For example, during busy events such as soccer matches, we've already encountered the case where a massive influx of users would cause the platform to become unavailable, at which point all the users would press F5 at the same time, drastically increasing the traffic on an already struggling platform. Moreover, requests are queued up and we end up answering requests issued 1 or 2 min earlier, and therefore answering people who have surely already left. 
+
+Jim mentions this issue, which has happened at amazon.com. 
+
+They were able to implement 2 solutions in particular. 
+
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:60]: - The first, when the servers can no longer respond, is to display a page with a message warning the user to wait a while and try again in a while. This had a surprisingly noticeable effect. Users were no longer repeatedly pressing the F5 button.
- Secondly, they have implemented mechanisms to detect massive retries and thus avoid transmitting traffic to their backend. In particular, they have implemented this in the WAF service.

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Secondly, they have implemented mechanisms to detect massive retries and thus avoid transmitting traffic to their backend. In particular, they have implemented this in the WAF service.
+
+At Bedrock Streaming we already display a page in case of trouble, but we can improve it to suggest to the user to wait before retrying. Moreover, we use Cloudfront and WAF on almost all our services. We have a few rules on WAF that allow us to deny traffic that seems illegitimate, but we're going to work on a new rule to avoid transmitting untimely user retries in the event of an overloaded system. 
+
+## Netflix caching 
+*“Who in this room is a netflix user?”* (90% of the room raises its hand): no doubt Prudhviraj Karumanchi, software engineer & Sriram Rangarajan, Senior Distributed Systems Engineer at Netflix, conference speakers, know how to introduce their talk and remind us that they are the market giants. [During their conference](https://www.youtube.com/watch?v=85TiFrDhCR4), they presented how Netflix uses the EVCache solution for multi-region cache replication.
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:68]: _(90% of the room raises its hand)_

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+![Quoting Marie Kondō, Netflix engineers remind us that one of their missions is to "spread joy"](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-netflix.jpg)
+
+This conference addressed many of the issues we are familiar with: a home page that calls on a large number of microservices to display content to the user, scaling issues, cost management...
+
+The heart of the conference detailed the architecture implemented by Netflix teams to replicate cache across multiple regions using EVCache, Kafka and SQS. 
+
+It was also interesting to see that at Netflix, as with us, it's important to build with costs in mind: after analyzing the costs of their inter-region traffic, they finally decided to remove their network load-balancer to make their architecture more cost-efficient.
+
+A long part of the conference was dedicated to the observability of the replication stack: our teams have also done a lot of work on this issue in recent years. While it's still interesting to see how Netflix does it, we don't have any new best practices to take away from this conference.
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:81]: Remplacer par : 
A long part of the conference was dedicated to the observability of the replication stack: our teams have also done a lot of work on this issue in recent years, so it was interesting to compare our practices on the subject of observability.

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+While we don't yet work on the same scale as Netflix, attending this conference allowed us to reinforce our idea that caching is essential in the architecture of a platform such as Bedrock Streaming. And it gives us new ways for reflection...
+
+## Takeaways from Reinvent2023 - The newcomers’ point of view
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:85]: Ne resort pas en titre dans ta preview ?

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+We were able to attend two of them: Adam Zelipsky's (CEO of aws) - for which the speakers were greeted by a fantastic rock band at 7:30 a.m., a guaranteed wake-up call - and Werner Vogels' (CPO & VP of amazon.com), where this time the welcome was provided by a fantastic string quartet.
+
+While Adam Zelisky's conference was undoubtedly a masterpiece (production, new services announced, influential clients such as Lidia Fonseca, chief digital and technology officer at Pfizer, speaking), it was Dr. Vogels' conference that impressed us the most.
+
+Based on [the laws of the "frugal architect](https://thefrugalarchitect.com/), this keynote spoke to everyone: technicians, business people, product managers... It took up the elementary concepts of what should motivate our design of IT solutions: cost awareness, the indispensable balance between commercial and technical needs, or the danger of never questioning oneself, quoting Grace Hopper: "One of the most dangerous phrases in the English language is: 'We've always done it this way:' 'We've always done it that way'."
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:101]: Based on [the laws of the "frugal architect"](https://thefrugalarchitect.com/)

Comment (BaronnePatPat) [reply _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:101]: Doublon : 'We've always done it that way'." + mettre en italique

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+We encourage you to watch [the replay of this keynote](https://www.youtube.com/watch?v=UTRBVPvzt9w), a must-see!
+
+![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-google.jpeg)
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:104]: ⚠️ pas la bonne photo, c'est celle de la sphère à nouveau :)

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+
+Beyond the gigantism of the event, let's look at what we've taken away from our participation in Reinvent. Attending a conference is always an opportunity to take a step back on our current work themes: we attended many conferences on resilience, scalability and new architectures: 
+it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...),
```
Comment (BaronnePatPat) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:108]: - it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...), 
- to transpose these themes into a context completely different from our own (“[1.5 million requests per second—a story from the Brazilian elections](https://www.youtube.com/watch?v=fGyvsNE4vh8)”) 
- or to appreciate the work we've done over the year, which sometimes goes beyond what's presented at conferences (“[Use new IAM Access Analyzer features on your journey to least privilege](https://www.youtube.com/watch?v=JpemUkU8INA)”)

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md (hunk 4 above)
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:68]:
```suggestion
*“Who in this room is a netflix user?” (90% of the room raises its hand)*: no doubt Prudhviraj Karumanchi, software engineer & Sriram Rangarajan, Senior Distributed Systems Engineer at Netflix, conference speakers, know how to introduce their talk and remind us that they are the market giants. [During their conference](https://www.youtube.com/watch?v=85TiFrDhCR4), they presented how Netflix uses the EVCache solution for multi-region cache replication.
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md (hunk 5 above)
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:81]:
```suggestion
A long part of the conference was dedicated to the observability of the replication stack: our teams have also done a lot of work on this issue in recent years, so it was interesting to compare our practices on the subject of observability.
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+We were able to attend two of them: Adam Zelipsky's (CEO of aws) - for which the speakers were greeted by a fantastic rock band at 7:30 a.m., a guaranteed wake-up call - and Werner Vogels' (CPO & VP of amazon.com), where this time the welcome was provided by a fantastic string quartet.
+
+While Adam Zelisky's conference was undoubtedly a masterpiece (production, new services announced, influential clients such as Lidia Fonseca, chief digital and technology officer at Pfizer, speaking), it was Dr. Vogels' conference that impressed us the most.
+
+Based on [the laws of the "frugal architect](https://thefrugalarchitect.com/), this keynote spoke to everyone: technicians, business people, product managers... It took up the elementary concepts of what should motivate our design of IT solutions: cost awareness, the indispensable balance between commercial and technical needs, or the danger of never questioning oneself, quoting Grace Hopper: "One of the most dangerous phrases in the English language is: 'We've always done it this way:' 'We've always done it that way'."
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:101]:
```suggestion
Based on [the laws of the "frugal architect"](https://thefrugalarchitect.com/), this keynote spoke to everyone: technicians, business people, product managers... It took up the elementary concepts of what should motivate our design of IT solutions: cost awareness, the indispensable balance between commercial and technical needs, or the danger of never questioning oneself, quoting Grace Hopper: "One of the most dangerous phrases in the English language is: *'We've always done it this way'*"
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+We encourage you to watch [the replay of this keynote](https://www.youtube.com/watch?v=UTRBVPvzt9w), a must-see!
+
+![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-google.jpeg)
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:104]:
```suggestion
![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-werner.jpg)
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+These examples show that congestion collapse is a real problem that requires constant attention and planning. 
+
+At bedrock we've also encountered congestion collapse. For example, during busy events such as soccer matches, we've already encountered the case where a massive influx of users would cause the platform to become unavailable, at which point all the users would press F5 at the same time, drastically increasing the traffic on an already struggling platform. Moreover, requests are queued up and we end up answering requests issued 1 or 2 min earlier, and therefore answering people who have surely already left. 
+
+Jim mentions this issue, which has happened at amazon.com. 
+
+They were able to implement 2 solutions in particular. 
+
+The first, when the servers can no longer respond, is to display a page with a message warning the user to wait a while and try again in a while. This had a surprisingly noticeable effect. Users were no longer repeatedly pressing the F5 button.
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:61]:
```suggestion
- The first, when the servers can no longer respond, is to display a page with a message warning the user to wait a while and try again in a while. This had a surprisingly noticeable effect. Users were no longer repeatedly pressing the F5 button.
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+The heart of the conference detailed the architecture implemented by Netflix teams to replicate cache across multiple regions using EVCache, Kafka and SQS. 
+
+It was also interesting to see that at Netflix, as with us, it's important to build with costs in mind: after analyzing the costs of their inter-region traffic, they finally decided to remove their network load-balancer to make their architecture more cost-efficient.
+
+A long part of the conference was dedicated to the observability of the replication stack: our teams have also done a lot of work on this issue in recent years. While it's still interesting to see how Netflix does it, we don't have any new best practices to take away from this conference.
+
+While we don't yet work on the same scale as Netflix, attending this conference allowed us to reinforce our idea that caching is essential in the architecture of a platform such as Bedrock Streaming. And it gives us new ways for reflection...
+
+## Takeaways from Reinvent2023 - The newcomers’ point of view
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:85]:
```suggestion
## Takeaways from Reinvent2023. The newcomers’ point of view
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+
+Based on [the laws of the "frugal architect"](https://thefrugalarchitect.com/), this keynote spoke to everyone: technicians, business people, product managers... It took up the elementary concepts of what should motivate our design of IT solutions: cost awareness, the indispensable balance between commercial and technical needs, or the danger of never questioning oneself, quoting Grace Hopper: "One of the most dangerous phrases in the English language is: *'We've always done it this way'*"
+We encourage you to watch [the replay of this keynote](https://www.youtube.com/watch?v=UTRBVPvzt9w), a must-see!
+
+![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-werner.jpg)
+
+
+Beyond the gigantism of the event, let's look at what we've taken away from our participation in Reinvent. Attending a conference is always an opportunity to take a step back on our current work themes: we attended many conferences on resilience, scalability and new architectures: 
+it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...),
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:108]:
```suggestion
- it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...),
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+We encourage you to watch [the replay of this keynote](https://www.youtube.com/watch?v=UTRBVPvzt9w), a must-see!
+
+![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-werner.jpg)
+
+
+Beyond the gigantism of the event, let's look at what we've taken away from our participation in Reinvent. Attending a conference is always an opportunity to take a step back on our current work themes: we attended many conferences on resilience, scalability and new architectures: 
+it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...), 
+to transpose these themes into a context completely different from our own (“[1.5 million requests per second—a story from the Brazilian elections](https://www.youtube.com/watch?v=fGyvsNE4vh8)”)
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:109]:
```suggestion
- to transpose these themes into a context completely different from our own (“[1.5 million requests per second—a story from the Brazilian elections](https://www.youtube.com/watch?v=fGyvsNE4vh8)”)
```

File: _posts/2023-12-18-aws-reinvent-lasvegas-2023.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+We encourage you to watch [the replay of this keynote](https://www.youtube.com/watch?v=UTRBVPvzt9w), a must-see!
+
+![Screenshot of one of the excellent edits made for the keynote of Werner Vogels](/images/posts/2023-12-18-aws-reinvent-lasvegas-2023/aws-reinvent-2023-werner.jpg)
+
+
+Beyond the gigantism of the event, let's look at what we've taken away from our participation in Reinvent. Attending a conference is always an opportunity to take a step back on our current work themes: we attended many conferences on resilience, scalability and new architectures: 
+it enabled us to compare our practices with those of major players in the sector (Prime, Spotify, Peacock...), 
+to transpose these themes into a context completely different from our own (“[1.5 million requests per second—a story from the Brazilian elections](https://www.youtube.com/watch?v=fGyvsNE4vh8)”) 
+or to appreciate the work we've done over the year, which sometimes goes beyond what's presented at conferences (“[Use new IAM Access Analyzer features on your journey to least privilege](https://www.youtube.com/watch?v=JpemUkU8INA)”)
```
Comment (Eriwyr) [inline _posts/2023-12-18-aws-reinvent-lasvegas-2023.md:110]:
```suggestion
- or to appreciate the work we've done over the year, which sometimes goes beyond what's presented at conferences (“[Use new IAM Access Analyzer features on your journey to least privilege](https://www.youtube.com/watch?v=JpemUkU8INA)”)
```

**Pull request merged**